### PR TITLE
Fix IMAP spam tab crash when mailbox contains thousands of messages

### PR DESCRIPTION
**Pull request by patrickuhlmann**
Previously, `Imap::Connector#fetch_mails` loaded all messages from the selected mailbox at once using a sequence-number range fetch, including the full RFC822 body of every message. With 13,000+ spam messages this caused the application to crash due to exhausting available memory, before Kaminari could even paginate the result.

Fix this by replacing the single bulk fetch with a two-phase approach within one IMAP connection:
1. `uid_search(["ALL"])` retrieves only message UIDs — a lightweight list of integers regardless of mailbox size.
2. UIDs are sorted descending (newest first) and sliced to the current page.
3. `uid_fetch` retrieves the full message content for only the ~25 messages on the current page.

The old bulk-fetch logic is retained as `fetch_all_mails`, used by `AppStatus::Mail` which must inspect every inbox message to detect overdue (stuck) deliveries.

Fixes: https://github.com/hitobito/hitobito/issues/3549

### CODE REVIEW

File: app/utils/imap/connector.rb [hunk 1]
```diff
@@ -68,6 +81,14 @@ def config(key)
 
   private
 
+  def fetch_mails_by_uids(page_uids)
+    return [] if page_uids.empty?
+
+    fetch_data = @imap.uid_fetch(page_uids, attributes)
+    mails = fetch_data.blank? ? [] : fetch_data.map { |mail| Imap::Mail.build(mail) }
+    mails.sort! { |a, b| b.date.to_i <=> a.date.to_i }
```
Comment (amaierhofer) [inline app/utils/imap/connector.rb]: Das sort hat bei meinem tests mit er pbs prod nicht immer  funktioniert, da der der `envelope` der Mail mitunter `nil` war. Ist wahrscheinlich ein edge case aber müssen wir abfangen. Am einfachsten direkt in der `Mail`

Und dann fände ich noch schön, wenn die `Mail` selbst sortierbar wäre, dort also den comparison operator `<=>` implementieren.

Comment (patrickuhlmann) [reply app/utils/imap/connector.rb]: Hast du schon versucht zu pushen. Hatte schon von Anfang an "Allow edits and access to secrets by maintainers" gesetzt auf dem PR:
image

Comment (amaierhofer) [reply app/utils/imap/connector.rb]: Stimmt. das Problem lag bei mir, push hat geklappt, danke!
